tools/scripts/firehose.py

```diff
@@ -129,17 +131,35 @@ def generate_jobs(jobs):
     from awx.main.models import UnifiedJob, Job, JobTemplate
     fields = list(set(Job._meta.fields) - set(UnifiedJob._meta.fields))
     job_field_names = set([f.attname for f in fields])
-    jt = JobTemplate.objects.first()
-    jt_defaults = dict(
-        (f.attname, getattr(jt, f.attname))
-        for f in JobTemplate._meta.get_fields()
-        if f.editable and f.attname in job_field_names and getattr(jt, f.attname)
-    )
-    jt_defaults['job_template_id'] = jt.pk
+    # extra unified job field names from base class
+    for field_name in ('name', 'created_by_id', 'modified_by_id'):
+        job_field_names.add(field_name)
+    jt_count = JobTemplate.objects.count()
+
+    def make_batch(N, jt_pos=0):
+        jt = None
```

Comment:
If you _truly_ want it to be random, something like this would be a lot simpler than tracking counts and doing modulo math:

`JobTemplate.objects.order_by("?").first()`

Reply:
Talked w/ @AlanCoding about this and at the scale of JT he's dealing with, that's slow.